Update and improve the API landing page, add more topics (#1376)
aiven/devportal: docs/tools/api.rst

Aiven API
=========

> Use the Aiven API to programmatically perform any task that you can do through the web interface. This is an ideal way to automate tasks involving Aiven at every stage of your workflow.

Reviewer: ```suggestion
Use the Aiven API to perform any programming task that you can perform through the web interface. This is an ideal way to automate tasks, involving Aiven at every stage of your workflow.
```

Reviewer: That just repeats the original text, but it clears up what "programmatically" means in this context :-D it's fine the way it was, then.

Common use cases for the Aiven API:

* Use with CI (Continuous Integration) to spin up data platforms on Aiven for use during test runs.

* Integrate with other parts of your existing automation setup to achieve complex tasks.

* Deploy and tear down development or demo platforms on a schedule.

> We make the API available to all Aiven users. It is also the engine behind our web console, so you should find that all operations are also available via API.

Reviewer: ```suggestion
We make the API available to all Aiven users. It is also the engine behind the Aiven Console, so you should find that all operations are also available through the API.
```

Reviewer: Assuming this is the Aiven Console you're talking about, Slab says we need to word it like this.

API quickstart
--------------

* **Postman**: Try `Aiven on Postman <https://www.postman.com/aiven-apis/workspace/aiven/documentation/21112408-1f6306ef-982e-49f8-bdae-4d9fdadbd6cd>`_ and start working with your data platform programmatically.

* **API documentation**: Check the `API documentation and OpenAPI description <https://api.aiven.io/doc/>`_ to work with the API directly.

* **Examples**: See the API in action with some :doc:`api/examples`.

Authentication
--------------

> Some endpoints (but not quite all) require authentication. You will need an authentication token from the `profile section of your Aiven console <https://console.aiven.io/profile/auth>`_.

Reviewer: ```suggestion
Some (but not quite all) endpoints require authentication. You'll need an authentication token from the `profile section of your Aiven console <https://console.aiven.io/profile/auth>`_.
```
